socorro/lib/util.py

```diff
@@ -144,3 +146,30 @@ def _retry_fun(*args, **kwargs):
         return _retry_fun
 
     return _retry_inner
+
+
+# Bug 1911612: Temporary decorator to log the method name, arguments and traceback
+# for each ESCrashStorage and SuperSearch method to better understand where ES gets
+# used by the webapp.
+def es_usage_logger(func):
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        stack_trace = "".join(traceback.format_stack())
+        class_name = self.__class__.__name__
+        method_name = func.__name__
+
+        log_entry_part_one = {
+            "bug_id": "1911612",
+            "method": f"{class_name}.{method_name}",
+            "arguments": {"args": args, "kwargs": kwargs},
+        }
+
+        log_entry_part_two = f"\ntraceback:\n{stack_trace}"
```

Comment:
I had to break this up into two parts, because `json.dumps` escapes newline characters, which I wanted to preserve in the traceback.

Open to a better way to make these logs more searchable/parseable downstream if others have better ideas!

Reply:
I don't think I understand what you're doing here. Are you planning to land these changes and have them run on stage and prod? Or are these changes limited to exploration in your local dev environment?

Reply:
I added an "Overall approach" paragraph to the [PR description](https://github.com/mozilla-services/socorro/pull/6707#issue-2512509905) that I think answers your question. Basically, the plan right now is to temporarily deploy the changes to prod to gather some usage data and then revert the change.

Reply:
I misunderstood the "This could ..." sentence as being theoretical and not the plan.

I think the underlying assumption for putting this in production is that the way production is used will organically uncover most/all possible uses of Elasticsearch by the code, that we'd be able to extract this usage information from the logs, and that this information will be more comprehensive than what we'd get out of running unit tests and test scenarios. I'm skeptical that this is compellingly true and I'd be hesitant to resort to this. I've made significant changes to the Elasticsearch code over the years by tracing the code in local dev and test environments and not doing something like this in production.

Have you tried capturing HTTP requests to Crash Stats and replaying them in a local dev environment? That continues this initiative, but without having to land and deploy anything.

Reply:
Thanks for the suggestion! That sounds like it could be lower risk/faster, though I'm not sure I've done something like that before. Do you have any tooling recommendations for doing this? I'll do a search as well after lunch.

Reply:
Oh, I primarily meant tooling for "capturing HTTP requests to Crash Stats and replaying them in a local dev environment", but I'm interested in any tooling recommendations in general for this approach.